packages/utils/src/dotenv.ts

```diff
@@ -0,0 +1,44 @@
+import * as dotenv from 'dotenv';
+import * as dotenvExpand from 'dotenv-expand';
+import * as fs from 'fs/promises';
+import path from 'path';
+import process from 'process';
+import { assert } from './assert';
+import { fromEntries } from './object';
+
+export function search(): string {
+  return path.resolve(process.cwd(), '.env');
+}
+
+export interface DotEnvParseArgs {
+  /**
+   * The filepath where the .env will be searched for
+   */
+  path: string;
+  /**
+   * Will filter resulting variables to only the ones starting with value of `filterPrefix`.
+   */
+  filterPrefix: string;
+}
+
+export async function parse(
+  { path, filterPrefix }: DotEnvParseArgs = {
+    path: search(),
+    filterPrefix: 'SNAP_',
```

Comment:
Maybe add this as a constant on top of this file.

```suggestion
    filterPrefix: DEFAULT_FILTER_PREFIX,
```